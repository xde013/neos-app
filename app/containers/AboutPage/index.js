/* eslint-disable jsx-a11y/anchor-is-valid */
/*
 * About Pages
 */
import React from 'react';
import { Helmet } from 'react-helmet';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import InfoIcon from '@material-ui/icons/Info';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';

export const AboutPage = () => {
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <>
      <Helmet>
        <title>About Page</title>
        <meta name="description" content="About page" />
      </Helmet>
      <Grid
        container
        direction="row"
        justify="center"
        alignItems="center"
        style={{ height: '75vh' }}
      >
        <Grid item xs={6}>
          <Paper variant="outlined" square style={{ background: '#e5e5e5' }}>
            <Box p={4}>
              <Typography variant="h2" gutterBottom>
                About <InfoIcon fontSize="large" />
              </Typography>
              <Typography variant="body1">
                This is a test project made with React focused on performance
                and best practices, showcasing{' '}
                <Link
                  variant="body1"
                  component="button"
                  color="primary"
                  underline="none"
                  onClick={handleClickOpen}
                >
                  NEOs (Near Earth Objects)
                </Link>{' '}
                data provided by
                <em> NASA</em>.
              </Typography>
            </Box>
          </Paper>
        </Grid>
      </Grid>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="info-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">What are NEOs?</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            Near-Earth objects (NEOs) are asteroids or comets of sizes ranging
            from metres to tens of kilometres that orbit the Sun and whose
            orbits come close to that of Earth&apos;s. Of the more than 600 000
            known asteroids in our Solar System, more than 20 000 are NEOs.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary" autoFocus>
            OK
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default AboutPage;
