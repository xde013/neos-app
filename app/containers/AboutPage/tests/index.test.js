import React from 'react';
import { render } from '@testing-library/react';

import AboutPage from '../index';

describe('<AboutPage />', () => {
  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(<AboutPage />);

    expect(firstChild).toMatchSnapshot();
  });
});
