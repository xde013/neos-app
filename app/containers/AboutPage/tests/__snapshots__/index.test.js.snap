// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> should render its heading 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  style="height: 75vh;"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <div
      class="MuiPaper-root MuiPaper-outlined"
      style="background: rgb(229, 229, 229);"
    >
      <div
        class="MuiBox-root MuiBox-root-133"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
        >
          About 
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          This is a test project made with React showcasing
           
          <button
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiLink-button MuiTypography-body1 MuiTypography-colorPrimary"
          >
            NEOs (Near Earth Objects)
          </button>
           
          data provided by
          <em>
             NASA
          </em>
          .
        </p>
      </div>
    </div>
  </div>
</div>
`;
